src/matrix.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include "spaces.hpp"
+//#include "stdx/traits.hpp"
+
+//#include <algorithm>
+//#include <array>
+//#include <cstddef>
+//#include <functional>
+//#include <iostream>
+//#include <numeric>
+//#include <ranges>
+//#include <utility>
+
+namespace opt {
+
+template <Arithmetic T, std::size_t Rows, std::size_t Cols>
+struct matrix {
+    static constexpr std::size_t rows = Rows;
+    static constexpr std::size_t cols = Cols;
+    using entries_type = T;
+    using columns_type = vector<T, Rows>;
+    using size_type = std::size_t;
+
+    std::array<vector<T, Rows>, Cols> columns{};
+
+    // Defaulted move constructors are auto noexcept
+    // NOLINTBEGIN(performance-noexcept-move-constructor)
+
+    matrix() = default;
```

Comment:
Are you defaulting everything? If so, just remove everything except the default constructor.

Reply:
Yes, I should follow the rule of 0